### PR TITLE
Resurrect _xpack/migration/deprecations routes

### PR DESCRIPTION
**Pull request by joegallo**
Related to #51816 / #68905.

Adds back the `_xpack/migration` routes when using rest compatibility for a request.

### CODE REVIEW

**Comment by joegallo** [comment]
See https://github.com/elastic/elasticsearch/pull/73930#issuecomment-857228008 re: `Team:Clients` -- I don't think there's anything the clients team needs to do here.

File: x-pack/plugin/build.gradle [hunk 1]
```diff
@@ -99,6 +99,9 @@ tasks.named("transformV7RestTests").configure({ task ->
   task.replaceKeyInDo("license.post_start_trial", "xpack-license.post_start_trial")
   task.addAllowedWarningRegex(".*_xpack/license.* is deprecated.*")
 
+  task.replaceKeyInDo("migration.deprecations", "xpack-migration.deprecations")
+  task.addAllowedWarningRegex(".*_xpack/migration.* is deprecated.*")
```
Comment (pgomulka) [inline x-pack/plugin/build.gradle:103]: maybe we could apply to the test that is using it?

**Review by pgomulka — APPROVED** [review]
LGTM,
left one question

File: x-pack/plugin/build.gradle (hunk 1 above)
Comment (joegallo) [reply x-pack/plugin/build.gradle:103]: With #75950, there's only one test that's using the change (`"Test get SSL certificates"`) so it's straightforward to limit ourselves to just the one test. Here, though, there's two (`"Test Deprecations"` and `"Test ml"`) so I don't think it's worth the additional lines of code to limit ourselves.

I'm going to merge this as is, but if you disagree strongly we can always tweak change it as a new PR against `master`.

**Pull request merged**